jormungandr/src/metrics/backends/prometheus_exporter.rs

```diff
@@ -0,0 +1,264 @@
+use crate::metrics::MetricsBackend;
+
+use chain_impl_mockchain::fragment::Fragment;
+use chain_impl_mockchain::transaction::Transaction;
+use chain_impl_mockchain::block::BlockContentHash;
+use chain_impl_mockchain::value::{Value, ValueError};
+
+use std::convert::TryInto;
+use std::time::SystemTime;
+use std::sync::Arc;
+
+use prometheus::core::{AtomicU64, GenericGauge};
+use prometheus::{Encoder, IntCounter, Registry, TextEncoder};
+use arc_swap::ArcSwapOption;
+
+type UIntGauge = GenericGauge<AtomicU64>;
+
+pub struct Prometheus {
+    registry: Registry,
+
+    tx_recv_cnt: IntCounter,
+    block_recv_cnt: IntCounter,
+    peer_connected_cnt: UIntGauge,
+    peer_quarantined_cnt: UIntGauge,
+    peer_available_cnt: UIntGauge,
+    peer_total_cnt: UIntGauge,
+    slot_start_time: UIntGauge,
+    block_tx_count: UIntGauge,
+    block_input_sum: UIntGauge,
+    block_fee_sum: UIntGauge,
+    block_content_size: UIntGauge,
+    block_epoch: UIntGauge,
+    block_slot: UIntGauge,
+    block_chain_length: UIntGauge,
+    block_time: UIntGauge,
+    block_hash: Vec<UIntGauge>,
+
+    block_hash_value: ArcSwapOption<BlockContentHash>,
+}
+
+impl Prometheus {
+    pub fn new() -> Self {
+        Default::default()
+    }
+
+    pub fn http_response(&self) -> Result<impl warp::Reply, warp::Rejection> {
+        if let Some(block_hash) = self.block_hash_value.load_full() {
```

Comment:
If I'm not wrong this resets the counters each time the function is called even if the values did not change. Maybe we could put a `None` back in the pointer so that successive calls do not do unnecessary work? (I would avoid using a swap as it would require a write each time)